docs/ossm/ossm2-migration/migration-checker.sh

```diff
@@ -192,39 +202,93 @@ check_proxies_updated() {
         local sanitized_latest_version
         sanitized_latest_version=$(cut -c2- <<< "$LATEST_VERSION")
         if [ "$sanitized_version" != "$sanitized_latest_version" ]; then
-            warning "pod: $name/$namespace is running a proxy at an older version: $sanitized_version Please update your ServiceMeshControlPlane to the latest version: ${LATEST_VERSION} and then restart this workload."
-            ((num_warnings+=1))
+            add_warning "pod: $name/$namespace is running a proxy at an older version: $sanitized_version Please update your ServiceMeshControlPlane to the latest version: ${LATEST_VERSION} and then restart this workload."
         fi
     done
 
-    if [ "$num_warnings" -gt 0 ]; then
-        ((TOTAL_WARNINGS += num_warnings))
-    else
-        success "All proxies are on the latest version."
-    fi
-
-    echo -e "$SPACER"
+    check_for_new_warnings $num_warnings "All proxies are on the latest version."
 }
 
 check_smcps() {
-    echo -e "Checking ServiceMeshControlPlanes...\n"
+    print_section "ServiceMeshControlPlanes"
     # Find smcps and check each one.
     # Format is name/namespace.
     for smcp in $(oc get smcp -A -o jsonpath='{range .items[*]}{.metadata.name}/{.metadata.namespace}{" "}{end}'); do
         IFS="/" read -r name namespace <<< "$smcp"
         check_smcp "$name" "$namespace"
     done
+}
 
+check_kiali() {
+    local name=$1
+    local namespace=$2
+
+    local num_warnings=$TOTAL_WARNINGS
+
+    echo -e "Kiali\nName: ${BLUE}$name${BLANK}\nNamespace: ${BLUE}$namespace${BLANK}\n"
+
+    local kiali
+    kiali=$(oc get kiali "$name" -n "$namespace" -o json)
+
+    local current_version
+    current_version=$(echo "$kiali" | jq -r '.spec.version')
+    if [[ "$current_version" != "$LATEST_KIALI_VERSION" && "$current_version" != "default" ]]; then
+        add_warning "Your Kiali is not on the latest version. Current version: '$current_version'. Latest version: '$LATEST_KIALI_VERSION'. Please upgrade your Kiali to the latest version."
+    fi
+
+    check_for_new_warnings $num_warnings "Kiali $name/$namespace is on the latest version."
     echo -e "$SPACER"
 }
 
+check_kialis() {
+    print_section "Kiali"
+
+    if ! oc get crds kialis.kiali.io > /dev/null 2>&1
+    then
+        echo "Kiali CRD is not detected. Skipping Kiali checks..."
+        return
+    fi
+    # Find smcps and check each one.
+    # Format is name/namespace.
+    for smcp in $(oc get kiali -A -o jsonpath='{range .items[*]}{.metadata.name}/{.metadata.namespace}{" "}{end}'); do
+        IFS="/" read -r name namespace <<< "$smcp"
+        check_kiali "$name" "$namespace"
+    done
+
+    local num_warnings=$TOTAL_WARNINGS
+
+    # Check Kiali operator
+    # Find kiali-ossm subscription and then use that to find the operator/csv namespace
+    local operator_namespace
+    operator_namespace=$(kubectl get subscriptions.operators.coreos.com -A -o jsonpath='{.items[?(@.metadata.name=="kiali-ossm")].metadata.namespace}')
```

Comment:
```suggestion
    operator_namespace=$(oc get subscriptions.operators.coreos.com -A -o jsonpath='{.items[?(@.metadata.name=="kiali-ossm")].metadata.namespace}')
```

Reply:
Agh I forgot to change all these! Thanks